src/json_schema_validation.rs

```diff
@@ -0,0 +1,344 @@
+use anyhow::{bail, Context, Result};
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::collections::HashMap;
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+#[serde(rename_all = "lowercase")]
+pub enum SchemaType {
+    String,
+    Number,
+    Integer,
+    Boolean,
+    Array,
+    Object,
+}
+
+/// Schema Validator is a JSON Schema descriptor used to evaluate the return value of a JsonPath
+/// expression, used by the presentation definition constraints field to ensure the property value
+/// meets the expected schema.
+///
+/// For more information, see the field constraints filter property:
+///
+/// - [https://identity.foundation/presentation-exchange/spec/v2.0.0/#input-descriptor-object](https://identity.foundation/presentation-exchange/spec/v2.0.0/#input-descriptor-object)
+///
+/// - [https://json-schema.org/understanding-json-schema](https://json-schema.org/understanding-json-schema)
+///
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct SchemaValidator {
```

Comment:
Add a `[const](https://json-schema.org/understanding-json-schema/reference/const)` value for equality check or restricted to a certain value. 

Add `[enum](https://json-schema.org/understanding-json-schema/reference/enum)` type to ensure `one-of` predicate for inclusion in a set.

Reply:
Ensure the validator matches the definitions specified here: https://json-schema.org/draft/2020-12/json-schema-validation

Reply:
This is handled in #9 and has been merged into this branch.